corehq/pillows/case_search.py

```diff
@@ -0,0 +1,151 @@
+from pillowtop.reindexer.reindexer import PillowReindexer
+
+
+class CaseSearchPillow(CasePillow):
+    # TODO: Remove this once elasticsearch gets bootstrapped with new pillows
+    # (Used in order to set up index)
+    """
+    Nested case properties indexer.
+    """
+    es_alias = CASE_SEARCH_ALIAS
+
+    es_index = CASE_SEARCH_INDEX
+    default_mapping = CASE_SEARCH_MAPPING
+
+    def change_trigger(self, changes_dict):
+        return
+
+
+def transform_case_for_elasticsearch(doc_dict):
+    doc = {
+        desired_property: doc_dict.get(desired_property)
+        for desired_property in CASE_SEARCH_MAPPING['properties'].keys()
+        if desired_property != 'case_properties'
+    }
+    doc['_id'] = doc_dict.get('_id')
+    doc['case_properties'] = _get_case_properties(doc_dict)
+    return doc
+
+
+def _get_case_properties(doc_dict):
+    base_case_properties = [
+        {'key': 'name', 'value': doc_dict.get('name')},
+        {'key': 'external_id', 'value': doc_dict.get('external_id')}
+    ]
+    dynamic_case_properties = [
+        {'key': key, 'value': value}
+        for key, value in doc_dict.iteritems()
+        if _is_dynamic_case_property(key)
+    ]
+
+    return base_case_properties + dynamic_case_properties
+
+
+def _is_dynamic_case_property(prop):
+    """
+    Finds whether {prop} is a dynamic property of CommCareCase. If so, it is likely a case property.
+    """
+    return not inspect.isdatadescriptor(getattr(CommCareCase, prop, None)) and re.search(r'^[a-zA-Z]', prop)
+
+
+class CaseSearchPillowProcessor(ElasticProcessor):
+    def process_change(self, pillow_instance, change, do_set_checkpoint):
+        domain = self._get_domain(change)
+        change_object = self._get_change_object(change)
+        if domain and case_search_enabled_for_domain(domain):
+            super(CaseSearchPillowProcessor, self).process_change(
+                pillow_instance, change_object, do_set_checkpoint
+            )
+
+    def _get_domain(self, change):
+        if isinstance(change, dict):
+            return change.get('key', [])[0]
+        elif isinstance(change, Change):
+            return change.metadata.domain
+
+    def _get_change_object(self, change):
+        if isinstance(change, dict):
+            return change_from_couch_row(change)
+        elif isinstance(change, Change):
+            return change
+
+
+def get_case_search_to_elasticsearch_pillow(pillow_id='CaseSearchToElasticsearchPillow'):
+    checkpoint = PillowCheckpoint(
+        'case-search-to-elasticsearch',
+    )
+    case_processor = CaseSearchPillowProcessor(
+        elasticseach=get_es_new(),
+        index_meta=ElasticsearchIndexMeta(index=CASE_SEARCH_INDEX, type=CASE_ES_TYPE),
+        doc_prep_fn=transform_case_for_elasticsearch
+    )
+    return ConstructedPillow(
+        name=pillow_id,
+        document_store=None,
+        checkpoint=checkpoint,
+        change_feed=KafkaChangeFeed(topics=[topics.CASE], group_id='cases-to-es'),
```

Comment:
I think this should also subscribe to `topics.CASE_SQL` and that might just work out of the box